src/modules/lightstyle.rs

```diff
@@ -118,6 +131,10 @@ pub fn on_cl_parse_lightstyle(marker: MainThreadMarker) {
     // Then, if we don't have any thing for our cvar, which is style is normal
     // and no custom. THen we just don't do anything.
     if BXT_LIGHTSTYLE.as_u64(marker) != 0 || !BXT_LIGHTSTYLE_CUSTOM.to_string(marker).is_empty() {
+        let cl_lightstyle = &mut unsafe { *engine::cl_lightstyle.get(marker) };
+
+        // More often a map's default lightstyle will be empty.
+        *ORIGINAL_LIGHTSTYLE.borrow_mut(marker) = cl_lightstyle[0].map.to_vec();
         apply_from_cvars(marker);
```

Comment:
This is holding a `&mut` reference on lightstyle while it is modified in `apply_from_cvars`, which is UB. Limit the scope of the `&mut`.

Reply:
Interesting.